Upgrade to Cordova 9

alalamav opened the pull request:
* Upgrades to cordova 9.0.0, cordova-ios 5.0.1, cordova-android 8.0.0, and cordova-browser 6.0.0.
* Later versions of Cordova are more tightly integrated with npm, which means package.json takes precedence over config.xml
  * Moves dependencies (i.e. plugins, platform engines) from config.xml to package.json so cordova doesn't modify package.json on every run.
  * Adds .npmrc to prevent package-lock.json from being created, as it may conflict with yarn.lock.
* cordova-android 8.0.0 supports target SDK 28. Removes two workarounds that support it in cordova 6.5.0.
* Verified there are no regressions in local builds or CI.

bemasc commented on package.json:
```diff
@@ -38,8 +43,11 @@
     "clang-format": "^1.0.55",
-    "cordova": "^6.5.0",
-    "cordova-custom-config": "^5.0.2",
+    "cordova": "~9.0.0",
```
Any reason not to use `^` for these, like we do with all the other dependencies?

alalamav replied:
We have seen breakages in the past due to unintended minor version updates. I believe it's safer to only update the patch.

alalamav commented:
Good catch, @bemasc. I was missing the `github:` when I attempted to use GitHub plugins. This causes Cordova to fetch the plugin from nmp and fail. I had to fix a mismatch in the name and ID of [cordova-webintent](https://github.com/cordova-misc/cordova-webintent); now released as v2.0.0.

I have removed the affected plugins from third_party and updated the configuration in dd5ba70. PTAL.

bemasc commented on cordova-plugin-outline/package.json:
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "cordova-plugin-outline",
+  "version": "1.0.0",
```
I wonder what the right value for "version" is in a case like this.  If this number doesn't actually increment when the code changes, "0.0.0" or "0" might be clearer.

alalamav replied:
I was wondering too. We declared version 1.0.0 in plugin.xml since the start. I don't think it matters in practice, but happy to change it if you feel strongly about it.

bemasc replied:
Just a suggestion.  If you agree, it's a one-character change, and easy to check that it still works.

If you prefer it the other way, that's fine.

alalamav replied:
I like the suggestion :) Done in e8310c4.
